components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2839,46 +2853,42 @@ private ProvisioningConnectorConfig createProvisioningConnectorConfig(String out
      * @return Federated authenticator config of the specified ID.
      */
     private FederatedAuthenticatorConfig updateFederatedAuthenticatorConfig(String federatedAuthenticatorId,
-                                                                            FederatedAuthenticatorPUTRequest
-                                                                                    authenticator) {
-
-        FederatedAuthenticatorConfig authConfig = new FederatedAuthenticatorConfig();
-        String authenticatorName = base64URLDecode(federatedAuthenticatorId);
-        authConfig.setName(authenticatorName);
-        authConfig.setDisplayName(getDisplayNameOfAuthenticator(authenticatorName));
-        authConfig.setEnabled(authenticator.getIsEnabled());
+                  FederatedAuthenticatorPUTRequest authenticator) throws IdentityProviderManagementClientException {
 
-        String definedByType = null;
-        if (authenticator.getDefinedBy() != null) {
-            definedByType = authenticator.getDefinedBy().toString();
+        String authenticatorName = getDecodedAuthenticatorName(federatedAuthenticatorId);
+        DefinedByType definedByType = resolveDefinedByTypeToUpdateFederatedAuthenticator(authenticatorName);
+        if (DefinedByType.SYSTEM == definedByType) {
+            validateAuthenticatorProperties(authenticatorName, authenticator.getProperties());
         }
```

Comment:
Same comment above. Given the factor is there we should be able to move this to the factory